torch/_dynamo/variables/nn_module.py

```diff
@@ -259,14 +268,31 @@ def has_key_in_generic_dict(self, tx: "InstructionTranslator", key):
     def _custom_getattr_fallback(self, base, tx, name, obj_source):
         """Check for a __getattr__ and handle it specially if it is implemented"""
         if object_has_getattribute(base):
-            unimplemented("torch.nn.Module with a custom __getattribute__ defined")
+            unimplemented_v2(
+                gb_type="torch.nn.Module with a custom __getattribute__ defined",
+                context=f"var_getattr {self} {name}",
+                explanation="Dynamo does not support checking key existence "
+                "on `nn.Module` instances that have a custom "
+                "`__getattribute__` method defined.",
+                hints=[
+                    "Avoid defining `__getattribute__` in your module. ",
+                    *graph_break_hints.SUPPORTABLE,
+                ],
+            )
 
         getattr_fn = get_custom_getattr(base, ignore_nn_module_getattr=True)
         if getattr_fn is None:
             return None
 
         if not isinstance(getattr_fn, types.FunctionType):
-            unimplemented("torch.nn.Module with a non-function custom __getattr__")
+            unimplemented_v2(
+                gb_type="torch.nn.Module with a non-function custom __getattr__",
+                context=f"var_getattr {self}",
+                explanation="Dynamo encountered a custom `__getattr__` method "
+                "on an `nn.Module` instance that is not a standard Python function. "
+                "Only function-based `__getattr__` methods can be traced.",
+                hints=[*graph_break_hints.USER_ERROR],
```

Comment:
I think this graph break would also trigger on C functions as well - in that case, this is not a user error then?